content/blog/experiences-with-ci-cd.pt-br.md

```diff
@@ -0,0 +1,476 @@
+
+Um desígnio que me motiva bastante ultimamente é o que a
+[WeaveWorks][weaveworks-gitops] decidiu chamar de GitOps; um fluxo de trabalho
+--- filosofia, diriam alguns --- resumido por "um repositório Git deveria ser a
+única fonte da verdade." Há várias possíveis definições, mas esta deve servir:
+
+> Um repositório Git é a única fonte da verdade para o estado desejado do
+> sistema inteiro. Este estado desejado é descrito de forma declarativa,
+> enquanto mecanismos de convergência são implantados para garantir que ele é
+> atingido.
+
+Em outras palavras, _mudanças na configuração_ são relacionadas 1:1 a _pull
+requests_ em um fluxo de trabalho GitOps.
+
+Entenda por _configuração_ o estado de um sistema, não arquivos de configuração
+e afins usados por software para alterar seu comportamento. Exclua daí dados
+persistentes produzidos pelo uso do sistema --- que, com sorte, não afetam o seu
+comportamento general, pois determinismo em software é bem imprevisível, ou ao
+menos dificilmente gerenciável.
+
+Apesar deste jeito de fazer deployment ter seus próprios defeitos ou em algum
+momento sofrer um declínio de popularidade por qualquer razão, ele oferece
+_agilidade_ relativa, seguindo fielmente a cultura DevOps e metodologias ágeis
+de desenvolvimento.
+
+[weaveworks-gitops]: https://www.weave.works/blog/what-is-gitops-really
+
+### Como eu acho que o alcancei no meu setup
+
+A WeaveWorks criou o termo descrevendo um fluxo de trabalho que circula clusters
+[Kubernetes][k8s]. Faz bastante sentido falar de descrições declarativas do
+estado almejado quando objetos Kubernetes têm todos o que chamamos de manifesto
+YAML. Algumas propriedades só existem quando o objeto em si está "no ar", como
+aquelas dentro de `{.status}`, mas tirando isso os objetos em si são
+praticamente idênticos aos manifestos que os geraram.
+
+É possível desginar um fluxo de trabalho como _GitOps_ quando ele não envolve
+algo como Kubernetes? Pensando um pouco e com uma mente aberta, contanto que o
+estado desejado seja declarado em vez de listarmos instruções a serem
+executadas --- há controvérsias em dizer que o Ansible satisfaz essa condição
+devido à relevância da ordem das tasks --- e hajam mecanismos de convergência,
+_dá para dizer que sim._
+
+Digamos que apesar da ordem de execução das tarefas ser de grande importância ao
+Ansible, e a ausência de um esquema de resolução de dependências --- como que o
+Ansible saberia que o pacote `certbot` precisa estar instalado para podermos
+usar o comando `certbot`? ---, playbooks são uma especificação declarativa do
+estado desejado para nossos hosts. _Então_ estaríamos certos em dizer que um
+fluxo de trabalho que usa Ansible como mecanismo de convergência constitui
+GitOps.
+
+Eu acredito que dá para ignorar as partes do estado do sistema não gerenciadas
+pelo Ansible, já que há "partes não contempladas" também em fluxos de trabalho
+rodeando Kubernetes (pense em autenticação ou `PersistentVolume`s). Sejamos
```

Comment:
Was it supposed to be "rodando" or "rodeando"?

Reply:
"Rodeando", but since it was unclear, I switched it to "envolvendo" in ead6a2d. Is that any better?